src/m_binbuf.c

```diff
@@ -487,24 +487,16 @@ int canvas_getdollarzero(void);
 static int binbuf_expanddollsym(const char *s, char *buf, t_atom *dollar0,
     int ac, const t_atom *av, int tonew)
 {
-    int argno = (int)atol(s);
-    int arglen = 0;
-    const char *cs = s;
-    char c = *cs;
+    char *cs;
+    int argno = (int)strtol(s, &cs, 10);
 
     *buf=0;
-    while (c && (c>='0') && (c<='9'))
-    {
-        c = *cs++;
-        arglen++;
-    }
-
-    if (cs==s)      /* invalid $-expansion (like "$bla") */
+    if (argno < 0 || cs == s)      /* invalid $-expansion (like "$bla") */
     {
         sprintf(buf, "$");
         return 0;
     }
-    else if (argno < 0 || argno > ac) /* undefined argument */
+    else if (argno > ac) /* undefined argument */
```

Comment:
It's been a while, so I don't recall why I messed around with these 'if' statements. It's possible they can be reverted back to what they originally were.

Reply:
That's good question. `$-1` can be considered either an "invalid $-expansion" or an "undefined (out-of-range) argument". When in doubt, I'd say just keep the original logic :)

Reply:
After a bit of debugging, it seems like it's basically impossible to specify a truly negative index. If a dash comes right after the dollar symbol, it will never get to this function. The only way argno < 1 is if a ridiculously large number is specified, at which point, I guess `strtol` returns -1 to indicate that it failed to fully parse the number.

Reply:
I might be thinking too much about what technically ends up happening instead of what should happen in principle. Maybe it should be, or will be possible for negative indexes to reach this function in the future, or maybe there are edge cases I haven't thought of where it can still happen, so I'll change the if statements back to what they were.